lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -676,6 +676,10 @@ sub calculate_result($) {
                 $important_overall = FAILED;
             }
             else {
+                # override important if it's passed
+                if ($important_overall && $important_overall eq PASSED) {
+                    $important_overall = SOFTFAILED;
+                }
                 $overall = FAILED;
```

Comment:
Suppose first module is unimportant and fails => `$overall` is FAILED, `$important_overall` is undef
Next module is important and passed => `$overall` if FAILED, `$important_overall` is PASSED
Then test finishes (or the rest of modules pass).
Result is then PASSED even when there is one failed module.

I think this update to result should be at the end of the calculate_result outside of the for cycle.